### DISCUSSION TITLE
add structuredClone support

### DISCUSSION THREAD

**Discussion by raptoria** [vercel/next.js#33189] [Ideas]
### Describe the feature you'd like to request

add structured clone support since all browsers have implemented this API in their nightly releases
related ticket: https://github.com/vercel/next.js/issues/31629

### Describe the solution you'd like

https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

### Describe alternatives you've considered

not using any polyfills at the moment

**Comment by ahmed-adly-khalil** [comment]
any luck here?

**Comment by LuchoMorla** [comment]
i could not run structuredClone at a jsx component, but i solve the issue in the Build -> structuredClone is not define, creating a module like a middleware to run structureClone() in a function:
deepCopy.js 
`
const deepClone = (array) => {
    return structuredClone(array);
};
export default deepClone;
`
finnally i import the module to the component jsx:
component.jsx

`
...
import deepCopy from '../middleware/deepClone';
...
const fetch = [];
const clone = deepCopy(fetch);
...
`

**Reply by Takzzg** [reply]
nice workaround, worked almost first try. only caveat: 
i'm working on a TypeScript proyect, and a `deepCopy.ts` file didn't work, but a `deepCopy.js` file did. Not sure why
also VScode included the `deepCopy.js` file automatically to `tsconfig.json` (explicitly since its the only `.js` file)

```
{
  ...
    "include": [
      ...
      "scripts/utilities/DeepCopy.js"
    ]
}
```

**Comment by GrantGryczan** [comment]
`structuredClone` is now supported by default in all major browsers, and core-js now has a polyfill for it. It should be supported by Next.js.

**Comment by eaallen** [comment]
Just installed the latest version of next. I do not have `structuredClone` support
